com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -749,42 +790,90 @@ private void OnTransformParentChanged()
         // we call CheckOrphanChildren() method and quickly iterate over OrphanChildren set and see if we can reparent/adopt one.
         internal static HashSet<NetworkObject> OrphanChildren = new HashSet<NetworkObject>();
 
-        internal bool ApplyNetworkParenting()
+        internal bool ApplyNetworkParenting(bool removeParent = false, bool ignoreNotSpawned = false)
         {
             if (!AutoObjectParentSync)
             {
                 return false;
             }
 
-            if (!IsSpawned)
+            // SPECIAL CASE:
+            // The ignoreNotSpawned is a special case scenario where a late joining client has joined
+            // and loaded one or more scenes that contain nested in-scene placed NetworkObject children
+            // yet the server's synchronization information does not indicate the NetworkObject in question
+            // has a parent. Under this scenario, we want to remove the parent before spawning and setting
+            // the transform values. This is the only scenario where the ignoreNotSpawned parameter is used.
+            if (!IsSpawned && !ignoreNotSpawned)
             {
                 return false;
             }
 
-            if (!m_IsReparented)
+            // Handle the first in-scene placed NetworkObject parenting scenarios. Once the m_LatestParent
+            // has been set, this will not be entered into again (i.e. the later code will be invoked and
+            // users will get notifications when the parent changes).
+            var isInScenePlaced = IsSceneObject.HasValue && IsSceneObject.Value;
+            if (transform.parent != null && !removeParent && !m_LatestParent.HasValue && isInScenePlaced)
             {
-                return true;
+                var parentNetworkObject = transform.parent.GetComponent<NetworkObject>();
+
+                // If the parent is a GameObject then preserve that hierarchy
```

Comment:
Nit... Comment says "if the parent is a GameObject", but the check is actually checking if it's a NetworkObject

Reply:
Ahh... yeah I need to be more clear... 
If the parent is just a GameObject (without a NetworkObject...which would mean parentNetworkObject is null)... but it could be rephrased to be less confusing indeed.